ticker/ticker.go

```diff
@@ -0,0 +1,131 @@
+package ticker
+
+import "time"
+
+// Ticker defines a resumable ticker interface, whose activity can be toggled to
+// free up resources during periods of inactivity.
+//
+// Example of resuming ticker:
+//
+//   ticker.Resume() // can remove to keep inactive at first
+//   defer ticker.Stop()
+//   for {
+//     select {
+//       case <-ticker.Tick():
+//         if shouldGoInactive {
+//           ticker.Pause()
+//           continue
+//         }
+//         ...
+//
+//       case <-otherEvent:
+//         ...
+//         if shouldGoActive && !ticker.IsActive() {
+//           ticker.Resume()
+//         }
+//     }
+//
+// NOTE: ONE DOES NOT SIMPLY assume that Tickers are safe for concurrent access.
+type Ticker interface {
+	// Ticks returns a read-only channel delivering ticks according to a
+	// prescribed interval. The value returned does not need to be the same
+	// channel, and may be nil.
+	//
+	// NOTE: Callers should assume that reads from Ticks() are stale after
+	// any invocations of Resume, Pause, or Stop.
+	Ticks() <-chan time.Time
+
+	// IsActive returns true if the channel returned from Ticks() is firing
+	// at the prescribed interval. A Ticker must always start in an inactive
+	// state.
+	IsActive() bool
+
+	// Resume starts or resumes the underlying ticker, such that Ticks()
+	// will fire at regular intervals. After calling Resume, IsActive()
+	// should return true until receiving a call to Pause or Stop.
+	Resume()
+
+	// Pause suspends the underlying ticker, such that Ticks() stops
+	// signaling at regular intervals. After calling Pause, IsActive()
+	// should return false until receiving an invocation of Resume.
+	//
+	// NOTE: It MUST be safe to call Pause at any time, and more than once
+	// successively.
+	Pause()
+
+	// Stop suspends the underlying ticker, such that Ticks() stops
+	// signaling at regular intervals,  and permanently frees up any
+	// remaining resources. After calling Stop, IsActive() should return
+	// false.
+	//
+	// NOTE: The behavior of a Ticker is undefined after calling Stop.
+	Stop()
```

Comment:
Do we need to distinguish pausing from stopping at this point? I think we should stick to `Start/Stop` as that should be enough for our use case.

Reply:
`Start`/`Stop` is enough, but the mock ticker has a goroutine that needs to be cleaned up. In order to handle forced ticks while paused, there needs to be a distinction between pausing, and actually tearing down the ticker